mail_tracking/controllers/discuss.py

```diff
@@ -5,7 +5,7 @@
 
 
 class MailTrackingDiscussController(DiscussController):
-    @http.route()
+    @http.route(type="json")
```

Comment:
I defined this again here because of the following traceback when running the tests:

>   Traceback (most recent call last):
>   File "/social/mail_tracking/tests/test_mail_tracking.py", line 714, in test_mail_init_messaging
>     res = controller.mail_init_messaging()
>   File "/odoo/odoo/http.py", line 698, in route_wrapper
>     if routing['type'] == 'http':  # _generate_routing_rules() ensures type is set
> KeyError: 'type'

The error corresponds to [this line ](https://github.com/odoo/odoo/blob/6bb16bf8818042b05ff8d40f852fa8e7d9eba92f/odoo/http.py#L698)... however, I've found that by doing this in the `setUp`:

> for _ in http._generate_routing_rules(["mail", "mail_tracking"], nodb_only=False):
>             pass

The error is no longer triggered, even without the explicit `type="json"` on the controller, I'll update the code shortly to remove it, thanks! :)